src/csharp/Grpc.Core/Internal/BatchContextSafeHandle.cs

```diff
@@ -86,6 +86,23 @@ public Metadata GetReceivedInitialMetadata()
             return MetadataArraySafeHandle.ReadMetadataFromPtrUnsafe(metadataArrayPtr);
         }
 
+        // Gets data of recv_initial_metadata completion without throwing an
+        // exception if the data is corrupt. This is to stop UnaryCall from hanging
+        // if there is some internal problem.
+        // See https://github.com/grpc/grpc/issues/29854
+        public Metadata TryGetReceivedInitialMetadata()
```

Comment:
I went through the thread of #29854 and didn't see any mention of failing to parse the InitalMetadata. The only issue that's reported is an error being thrown when GetReceivedStatusOnClient tries to convert the statusDetails to string. So it seems that proactively trying to modify GetReceivedInitialMetadata is unnecessary?

The location of the problem:
https://github.com/grpc/grpc/blob/90b0ea387e988b8615882e152a5152ed0801ec06/src/csharp/Grpc.Core/Internal/BatchContextSafeHandle.cs#L94

Reply:
I was being thorough and handling all cases that could cause a hang if an exception was thrown